intents_test.go

```diff
@@ -269,34 +269,34 @@ func TestSubmitHandler(t *testing.T) {
 				Intents: []*Intent{
 					{
 						Sender: senderAddress,
-						From: Supply{
-							Type:     SupplyType,
-							Address:  "0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2",
-							Currency: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
-						},
-						To: Asset{
+						From: Asset{
 							Type:    TokenType,
 							Address: "0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2",
 							Amount:  "104",
 							ChainId: "1",
 						},
+						To: LoanTo{
+							Type:    LoanType,
+							Address: "0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2",
+							Asset:   "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+						},
 						Status: Received,
 					},
 				},
 			},
 			expectCode: http.StatusOK,
 		},
 		{
-			description: "Valid operation - Withdraw supply",
+			description: "jfkf",
 			payload: Body{
 				Intents: []*Intent{
 					{
 						Sender: senderAddress,
-						From: WithdrawSupply{
-							Type:     WithdrawSupplyType,
-							Address:  "0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2",
-							Currency: "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
-							Amount:   "104",
+						From: LoanFrom{
+							Type:    LoanType,
+							Address: "0x87870Bca3F3fD6335C3F4ce8392D69350B4fA4E2",
+							Asset:   "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",
+							Amount:  "104",
 						},
 						To: Asset{
 							Type:    TokenType,
```

Comment:
What about Approve pre Supply?

Reply:
this is just a test to make sure we can validate the contract addresses and amount